Saturation/SaturationAlgorithm.cpp

```diff
@@ -1300,20 +1316,18 @@ void SaturationAlgorithm::doOneAlgorithmStep()
  */
 MainLoopResult SaturationAlgorithm::runImpl()
 {
-  unsigned l = 0;
-
   // could be more precise, but we don't care too much
   unsigned startTime = Timer::elapsedDeciseconds();
   try {
-    for (;; l++) {
-      if (_activationLimit && l > _activationLimit) {
+    for (unsigned iter = 0;; iter++) {
+      if (_activationLimit && iter > _activationLimit) {
         throw ActivationLimitExceededException();
       }
       if(_softTimeLimit && Timer::elapsedDeciseconds() - startTime > _softTimeLimit)
         throw TimeLimitExceededException();
 
-      doOneAlgorithmStep();
-      env.statistics->activations = l;
+      doOneAlgorithmStep(iter);
+      env.statistics->activations = iter;
```

Comment:
Couldn't `doOneAlgorithmStep` update `env.statistics->activations` and thereby remove `iter` altogether?

Reply:
Good idea!